bot/receiver.go

```diff
@@ -30,82 +31,94 @@ func playAudioCmd(ctx CommandCtx) error {
 			log.Printf("Error joining VC to play audio: %s", err)
 			return ctx.Reply("Failed to join voice channel.")
 		}
-
+		ar := NewAudioReceiver(voiceConn)
+		activeReceiver = ar
 		go monitorVoiceActivity(s, gId, vc)
-		go StartAudioReceiver(voiceConn)
+		go ar.Start()
 		return ctx.Reply("Venova has entered the chat...")
 	}
 	return nil
 }
 
+func NewAudioReceiver(vc *discordgo.VoiceConnection) *AudioReceiver {
+	return &AudioReceiver{
+		StopChan: make(chan bool, 1),
+		Vc:       vc,
+	}
+}
+
 func gatherVars() string {
```

Comment:
This function feels kind of ugly for a few reasons:

- `gatherVars()` is objectively a bad name, it should be something like `getAudioServerPort()`
- Why is this function responsible for doing a global-level action like loading from a dot env file? Shouldn't that be done as an initialization step before this code can ever run?
- Why don't we just pass in `audioServerPort` as an arg to `NewAudioReceiver`? If you did the larger refactor I wanted above this would fit in cleanly, otherwise you could just store it as a private field on the struct and use it later in `Start()`